alicloud/data_source_alicloud_emr_instance_types.go

```diff
@@ -0,0 +1,169 @@
+package alicloud
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/emr"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func dataSourceAlicloudEmrInstanceTypes() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudEmrInstanceTypesRead,
+
+		Schema: map[string]*schema.Schema{
+			"destination_resource": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					"Zone",
+					"Network",
+					"InstanceType",
+					"SystemDisk",
+					"DataDisk",
+				}),
+			},
+			"cluster_type": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Does the `cluster_type` have valid values?

Reply:
Just like create emr clusters with resource, there is no need to valid cluster_type, cause the enumeration type of a cluster is dynamic in future to support other emr instance.